server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -932,6 +934,15 @@ protected Node(
                         .toInstance(new PeerRecoverySourceService(transportService, indicesService, recoverySettings));
                     b.bind(PeerRecoveryTargetService.class)
                         .toInstance(new PeerRecoveryTargetService(threadPool, transportService, recoverySettings, clusterService));
+                    b.bind(SegmentReplicationTargetService.class)
```

Comment:
if you are binding here, you do not need @Inject on `SegmentReplicationTargetService`'s constructor.

Reply:
Shouldn't this binding also be gated by the feature flag, like `SegmentReplicationSourceService` below?

Reply:
Yes removed `@Inject`  `SegmentReplicationTargetService`'s constructor. Yes this should also be gated by feature flag. But need some changes in binding SegmentReplicationCheckpointPublisher in IndicesModule, will add them in next commit.